packages/framework/src/Model/Mail/Mailer.php

```diff
@@ -5,20 +5,54 @@
 use Swift_Attachment;
 use Swift_Mailer;
 use Swift_Message;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 class Mailer
 {
+    /**
+     * @var \Symfony\Component\HttpKernel\KernelInterface
+     */
+    protected $kernel;
+
     /**
      * @var \Swift_Mailer
      */
     protected $swiftMailer;
 
     /**
      * @param \Swift_Mailer $swiftMailer
+     * @param \Symfony\Component\HttpKernel\KernelInterface $kernel
      */
-    public function __construct(Swift_Mailer $swiftMailer)
+    public function __construct(Swift_Mailer $swiftMailer, KernelInterface $kernel)
     {
         $this->swiftMailer = $swiftMailer;
+        $this->kernel = $kernel;
+    }
+
+    public function flushSpoolQueue()
+    {
+        $container = $this->kernel->getContainer();
+        if (!$container->has('mailer')) {
+            return;
+        }
+        $mailers = array_keys($container->getParameter('swiftmailer.mailers'));
```

Comment:
do you need to get only mailer names? isn't there a way to get the whole mailers? so you don't need to get them later on?

Reply:
i tried to call the same thing that is called on the releasing of spool. please look into the `vendor/symfony/swiftmailer-bundle/EventListener/EmailSenderListener.php`